Sources/SwiftDocC/Infrastructure/DocumentationCurator.swift

```diff
@@ -231,6 +231,21 @@ struct DocumentationCurator {
                     (childDocumentationNode.kind == .article || childDocumentationNode.kind.isSymbol || childDocumentationNode.kind == .tutorial || childDocumentationNode.kind == .tutorialArticle) else {
                         continue
                 }
+                
+                // A solution that suggest removing the list item that contain this link
+                var removeListItemSolutions: [Solution] {
+                    // Traverse the markup parents up to the nearest list item
+                    guard let listItem = sequence(first: link as (any Markup), next: \.parent).mapFirst(where: { $0 as? ListItem }),
+                          let listItemRange = listItem.range
+                    else {
+                        return []
```

Comment:
Is it possible for `removeListItemSolutions` to return an empty array? Shouldn't there always be a list item parent? Is there any other way to create a topic hierarchy?

That is, should we change `return []` to an assertion (in debug mode at least) since it must indicate some kind of programming error and not a valid content scenario?

Reply:
I added an assertion in https://github.com/swiftlang/swift-docc/pull/984/commits/a2c52e0d4a12774d5d7c047efe16d2ac86c17441